Lib/test/test_patma.py

```diff
@@ -2172,7 +2172,7 @@ def test_patma_204(self):
         def f(w):
             match w:
                 case 42:
-                    out = locals()
+                    out = {k : v for k,v in locals().items() if not k.startswith('$')}
```

Comment:
I wonder if the exceprion for `$` should be implemented in `locals()` rather than in its callers?

Reply:
I don’t know if we want to keep the $ vars. we did this because I couldn’t get it to work with a dedicated range for temps, but now we fixed it for consts so it should work.

Reply:
We might want these for debugging purposes. Perhaps `locals`/`frame.f_locals` should omit them, and we add `frame._f_registers` (or something) which is a list of temporaries?

Reply:
Maybe relevant: the unnamed iterator used in comprehensions is currently visible:

```py
>>> [locals() for _ in [None]]
[{'.0': <tuple_iterator object at 0x7f1a741370a0>, '_': None}]
```

Reply:
The `$` registers will be visible in a lot more cases than the `.0` iterator, and they are a detail of the implementation that will mostly just confuse users. If we're going to pull the consistency card I'd vote for not revealing that iterator either.

I could live with the registers showing in `f_locals` but not in `locals()` -- frame attributes are more of an implementation detail anyway.

Reply:
I made the change to filter locals(), but that's no good it breaks `locals()["x"] = 43`.

(see test.test_scope.ScopeTests.testClassNamespaceOverridesClosure)

Reply:
Why are we adding all these "$" variables to the locals?
We don't expose the stack, why expose the temporaries?

It looks like a considerable amount of the changes to both and tests are supporting/working around these additional, artificial local variables.